### PR TITLE
Automate ETL on Heroku

### PR DESCRIPTION
**Pull request by eeeschwartz**
* Extracts xls from Health Dept site
* Transforms to LIVES format
* Loads to GitHub (a separate repo for the LIVES data, currently under [my account](https://github.com/eeeschwartz/lexingtonky-lives-data))

The git/rugged code could stand to be its own class but this works in the meantime (it's tested on Heroku). 

Next step would be:
* move the data repo under OpenLex
* give it the CNAME lives.openlexington.org, so Yelp can pick it up
* set up a scheduled task on Heroku to run `bundle exec ruby lives_etl.rb` (once a day perhaps)

Feedback/improvements welcome.

### CODE REVIEW

**Comment by chaserx** [comment]
FUTURE:  we should think about some alert, dead man switch if something goes wrong in these automated steps.

File: lex_to_lives.rb [hunk 1]
```diff
@@ -71,6 +73,17 @@ def csv_parse(infile)
...
+  def feed_info
+    info = FeedInfo.new.tap do |f|
+      f.feed_date         = Date.today.strftime('%Y%m%d')
+      f.feed_version      = '2.0'
+      f.municipality_name = 'Lexington-Fayette County'
+      f.municipality_url  = 'http://lexingtonhealthdepartment.org/ProgramsServices/RestaurantInspections/tabid/87/Default.aspx'
+      f.contact_email     = 'healthscores@openexington.org'
+    end
+    [info]
+  end
```
Comment (chaserx) [inline lex_to_lives.rb]: I forget about `#tap`

File: lives_etl.rb [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+  Rugged::Commit.create(repo,
+    author: commit_author,
+    committer: commit_author,
+    message: 'Automatic feed update',
+    parents: [repo.head.target],
+    tree: commit_tree,
+    update_ref: 'HEAD')
+
+  credentials = Rugged::Credentials::UserPassword.new(username: ENV['GITHUB_USER'],
+    password: ENV['GITHUB_PASS'])
```
Comment (chaserx) [inline lives_etl.rb]: I think if you do `ENV.fetch('key')` then you get `KeyError` exception if the key is missing.

File: lives_etl.rb [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+  end
+end
+
+def csv_write(output_file, row_headers, spreadsheet)
+  csv_opts = {headers: row_headers, write_headers: true}
+
+  CSV.open(output_file, "wb", csv_opts) do |csv|
+    for i in 2..spreadsheet.last_row do
+      csv << spreadsheet.row(i)
+    end
```
Comment (chaserx) [inline lives_etl.rb]: is this more rubyish

```ruby
(2..spreadsheet.last_row).each do |row|
  csv << spreadsheet.row(row)
end
```

File: lives_etl.rb [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+end
+
+def stage_changes(repo)
+  should_push = false
+  repo.index.diff.each_delta do |d|
+    # only push if scores have changed
+    if (d.old_file[:path] == 'most_recent_food_scores.csv')
+      should_push = true
+    end
+
```
Comment (chaserx) [inline lives_etl.rb]: I'm not sure the parens are needed in the conditional...

**Comment by chaserx** [comment]
+1

**Comment by eeeschwartz** [comment]
@chaserx good calls, shake the copy pasta out of this bad boy.

**Pull request merged**